src/components/forms/Textarea/Textarea.tsx

```diff
@@ -0,0 +1,158 @@
+   */
+  autoExpand?: boolean;
+}
+
+const Textarea: React.FC<TextareaProps & UtrechtTextareaProps> = ({
+  name,
+  label = '',
+  isRequired = false,
+  description = '',
+  isDisabled = false,
+  autoExpand = false,
+  placeholder,
+  tooltip,
+  ...extraProps
+}) => {
+  const {validateField} = useFormikContext();
+  const [{value, ...props}, {error = '', touched}] = useField<string | undefined>({
+    name,
+    type: 'text',
+  });
+  const id = useId();
+  const textareaRef = useRef<HTMLTextAreaElement | null>(null);
+
+  const invalid = touched && !!error;
+  const errorMessageId = invalid ? `${id}-error-message` : undefined;
+
+  // Used in conjunction with `autoExpand` to calculate the textarea height
+  const numberOfLineBreaks = ((value || '').match(/\n/g) || []).length;
+
+  useLayoutEffect(() => {
+    const textarea = textareaRef.current;
+    if (!textarea || !autoExpand) {
+      return;
+    }
+
+    const calculateTextareaHeight = (rows: number): number => {
+      // Style values are returned as px values.
+      const {lineHeight, paddingBlockStart, paddingBlockEnd, borderBlockStart, borderBlockEnd} =
+        getComputedStyle(textarea);
+
+      // Parse all style values to float, to remove the trailing "px", and make them calculable.
+      return (
+        rows * parseFloat(lineHeight) +
+        parseFloat(paddingBlockStart) +
+        parseFloat(paddingBlockEnd) +
+        parseFloat(borderBlockStart) +
+        parseFloat(borderBlockEnd)
+      );
+    };
+
+    const resizeTextarea = () => {
+      // Include the line without linebreak in the calculation
+      const numberOfLines = numberOfLineBreaks + 1;
+
+      const minimalHeight: number = calculateTextareaHeight(textarea.rows);
+      const requiredHeight: number = calculateTextareaHeight(numberOfLines);
+
+      const newHeight = Math.max(minimalHeight, requiredHeight);
+      // Only update if new height is different from the current height
+      if (parseFloat(getComputedStyle(textarea).height) !== newHeight) {
+        textarea.style.height = `${newHeight}px`;
+      }
+    };
+
+    resizeTextarea();
+  }, [numberOfLineBreaks, autoExpand]);
+
+  return (
+    <FormField type="text" invalid={invalid} className="utrecht-form-field--openforms">
+      <Label
+        id={id}
+        isRequired={isRequired}
+        isDisabled={isDisabled}
+        tooltip={tooltip ? <Tooltip>{tooltip}</Tooltip> : undefined}
+      >
+        {label}
+      </Label>
+      <Paragraph>
+        <UtrechtTextarea
+          ref={textareaRef}
+          // ensure unsetting values doesn't ping-pong us between controlled/uncontrolled
+          // components
+          value={value ?? ''}
+          {...props}
+          onBlur={async e => {
+            props.onBlur(e);
+            await validateField(name);
+          }}
+          className={clsx('utrecht-textarea--openforms', {
+            'utrecht-textarea--openforms--no-resize': autoExpand,
```

Comment:
```suggestion
            'utrecht-textarea--openforms-no-resize': autoExpand,
```
probably best for namespace reasons

in BEM naming, the `__` and `--` separators can never repeat - you have a max of 3 levels deep (`block__element--modifier`)